rmf_site_editor/src/workspace.rs

```diff
@@ -264,211 +233,281 @@ pub fn dispatch_new_workspace_events(
+                        None
+                    }
+                    #[cfg(target_arch = "wasm32")]
+                    {
+                        let data =
+                            WorkspaceData::LoadSite(LoadSite::blank_L1("blank".to_owned(), None));
+                        Some(LoadWorkspaceFile(None, data))
+                    }
+                })
+                .then(process_load_files)
+                .connect(scope.terminate)
+        });
+
+        let load_workspace_from_path = world.spawn_workflow(|scope, builder| {
+            scope
+                .input
+                .chain(builder)
+                .map_async(|path| async move {
+                    let Some(data) = std::fs::read(&path)
+                        .ok()
+                        .and_then(|data| WorkspaceData::new(&path, data))
+                    else {
+                        warn!("Unable to read file [{path:?}] so it cannot be loaded");
+                        return None;
+                    };
+                    Some(LoadWorkspaceFile(Some(path.clone()), data))
+                })
+                .then(process_load_files)
+                .connect(scope.terminate)
+        });
+
+        let load_workspace_from_data = world.spawn_workflow(|scope, builder| {
+            scope
+                .input
+                .chain(builder)
+                .map_block(|data| Some(LoadWorkspaceFile(None, data)))
+                .then(process_load_files)
+                .connect(scope.terminate)
+        });
+
+        Self {
+            load_workspace_from_dialog,
+            create_empty_workspace_from_dialog,
+            load_workspace_from_path,
+            load_workspace_from_data,
         }
     }
 }
 
+impl<'w, 's> WorkspaceLoadingParams<'w, 's> {
+    /// Request to spawn a dialog and load a workspace
+    pub fn load_from_dialog(&mut self) {
+        self.commands
+            .request((), self.workspace_loading.load_workspace_from_dialog)
+            .detach();
+    }
+
+    /// Request to spawn a dialog to select a file and create a new site with a blank level
+    pub fn create_empty_from_dialog(&mut self) {
+        self.commands
+            .request(
+                (),
+                self.workspace_loading.create_empty_workspace_from_dialog,
+            )
+            .detach();
+    }
+
+    /// Request to load a workspace from a path
+    pub fn load_from_path(&mut self, path: PathBuf) {
+        self.commands
+            .request(path, self.workspace_loading.load_workspace_from_path)
+            .detach();
+    }
+
+    /// Request to load a workspace from data
+    pub fn load_from_data(&mut self, data: WorkspaceData) {
+        self.commands
+            .request(data, self.workspace_loading.load_workspace_from_data)
+            .detach();
+    }
+}
+
+/// `SystemParam` used to request for workspace loading operations
+#[derive(SystemParam)]
+pub struct WorkspaceLoadingParams<'w, 's> {
```

Comment:
Super super minor nitpick but what if we call this `WorkspaceLoader` instead of `WorkspaceLoadingParams`? I think `WorkspaceLoader` would make its purpose more clear. Users shouldn't really have to care that it's a struct of system params.

Reply:
https://github.com/open-rmf/rmf_site/pull/233/commits/41f6233fad7df6db384880798c8b518a3d0d9577